docs/specs/toc.yml

```diff
@@ -1079,3 +1079,85 @@ outputs:
         { "toc_title":"Title","href":"a#bookmark","monikers":[] }
       ]
     }
+---
+# uid reference
+inputs:
+  docfx.yml:
+  docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    ---
+  docs/TOC.yml: |
+    - name: Uid Ref
+      uid: a
+outputs:
+  docs/a.json:
+  docs/toc.json: |
+    {
+      "items": [
+        { "toc_title":"Uid Ref","href":"a","monikers":[] }
+      ]
+    }
+---
+# uid reference first
+inputs:
+  docfx.yml:
+  docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    ---
+  docs/b.md:
+  docs/TOC.yml: |
+    - name: Uid Ref
+      uid: a
+      href: b.md
```

Comment:
should we report a warning to user that the href will not work?

Reply:
Just checked v2 code, v2 does not log a warning in this case, but it will use href if uid-not-found.